cider/exec/nextgen/Parser.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_EXEC_NEXTGEN_PARSER_H
+#define CIDER_EXEC_NEXTGEN_PARSER_H
+
+#include <memory>
+
+#include "exec/nextgen/OpPipeline.h"
+#include "exec/template/RelAlgExecutionUnit.h"
+
+namespace cider::exec::nextgen {
+
+class Parser {
+ public:
+  // source--> filter -->sink
+  static std::unique_ptr<OpPipeline> toOpPipeline(const RelAlgExecutionUnit& eu) {
+    OpNodeVector nodes;
```

Comment:
`nodes = ConstructNodes(eu)`? So actually, we add an internal layer of `PlanNode `representation inside Cider and in future these nodes should be directly mapped to `Substrait::Rel` with `RelAlgExecutionUnit`. So do we need to design these nodes based on contents of corresponding substrait representation?